src/Network/Response.php

```diff
@@ -1245,10 +1287,10 @@ public function protocol($protocol = null)
     public function length($bytes = null)
     {
         if ($bytes !== null) {
-            $this->_headers['Content-Length'] = $bytes;
+            $this->_headers['content-length'] = $bytes;
         }
-        if (isset($this->_headers['Content-Length'])) {
-            return $this->_headers['Content-Length'];
+        if (isset($this->_headers['content-length'])) {
+            return $this->_headers['content-length'];
```

Comment:
Internally we might want 2 structures for headers. The first would be a normalized hash of names => values e.g

```php
[
  'content-type' => 'text/html'
]
```

The second would preserve the casing on headers:

```php
[
  'content-type' => 'Content-Type'
]
```

Having both would let us keep header operations simple, and allow us to preserve the developers desired header casing.